build.gradle

```diff
@@ -1216,12 +1237,16 @@ project(':clients') {
     testRuntimeOnly libs.jacksonJDK8Datatypes
     testImplementation libs.jose4j
     testImplementation libs.jacksonJaxrsJsonProvider
+
+    generator project(':generator')
   }
 
   task createVersionFile() {
-    ext.receiptFile = file("$buildDir/kafka/$buildVersionFileName")
+    def receiptFile = file("$buildDir/kafka/$buildVersionFileName")
+    inputs.property "commitId", commitId
```

Comment:
Can you clarify how why this is needed? I don't see any deletions on the left side that this would be replacing.

Reply:
Without declaring `commitId` and `version` as inputs to these `createVersionFile` tasks, they will never be UP-TO-DATE and won't be cacheable, so they will always run, even when they would just produce the exact same outputs. See: [Gradle Incremental Build docs](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:task_inputs_outputs).